### ISSUE TITLE
Check URL against predetermined value

### ISSUE THREAD

**Issue by nicholaswkc34** [selenide/selenide#2392]
## The problem
Dear all, I execute my test against some website but it does have random advertisement pop up but I able to predict its behaviours where it will only display advertisement when URL becomes www.nst.com.my/#google_vignette, so I will close it with the click. 

Questions: 
1. How to check URL becames certail value with polling interval?
2. How to close google chrome pop up notification with polling interval? with the allow and block button on left top corner. 

![IMG_20230729_060001-min](https://github.com/selenide/selenide/assets/44965697/28fd5791-0c91-40e2-87c8-1c056ff2b056)

## Tell us about your environment
* **Selenide Version**: 6.15.0
* **Chrome\Firefox\IE Version**: 113
* **Browser Driver Version**:113
* **Selenide-Appium Version**:-
* **IOS\Android\WEB version**:-
* **Selenium Version**: 4.9.1
* **OS Version**: Win 10

**Comment by nicholaswkc34** [comment]
WebDriverRunner.url() is the method to cehck URL but it's does not have poolin interval. How?

**Comment by asolntsev** [comment]
@nicholaswkc34 We don't know how to click "Allow" or "Block" in that dialog. I guess we cannot. 

Usually we just run chromedriver with some specific option that allows or block notification. You need to google for that options.


P.S. 
---------
About checking URL - we have two methods available:
1. `WebDriverRunner.url()` - returns current URL, doesn't wait for anything.
2. `webdriver().shouldHave(url("http://www.nst.com.my/#google_vignette"), Duration.ofSeconds(42));` - waits for given URL. Throws an exception if actual URL was not the expected one.

I guess none of them does perfectly what you want. 

But the problem is with what you want. :) 
Basically, you want to add an IF to your tests which has some kind of built-in waiting mechanism. But you want it to "wait" only in some cases, not always. This IF will slow down your tests. 

See more here: https://selenide.org/2019/12/02/advent-calendar-how-to-abuse-selenide/
